### PR TITLE
build: update to TypeScript 5.6 RC

### PR DESCRIPTION
**Pull request by crisbeto**
Updates the repo to the TypeScript 5.6 RC and narrows down the version to drop support for 5.4, in line with the compiler.

### CODE REVIEW

File: packages/angular_devkit/schematics/src/tree/action.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ export class ActionList implements Iterable<Action> {
   get length() {
     return this._actions.length;
   }
-  [Symbol.iterator]() {
+  [Symbol.iterator](): IterableIterator<Action> {
```
Comment (crisbeto) [inline packages/angular_devkit/schematics/src/tree/action.ts:153]: This needed an explicit return type or the API extractor would crash.

File: .ng-dev/format.mjs [hunk 2]
```diff
@@ -5,7 +5,12 @@
  */
 export const format = {
   'prettier': {
-    matchers: ['**/*.{mts,cts,ts,mjs,cjs,js,json,yml,yaml,md}'],
+    matchers: [
+      '**/*.{mts,cts,ts,mjs,cjs,js,json,yml,yaml,md}',
+
+      // Don't expect vendored files to be formatted.
+      '!packages/schematics/angular/third_party/**',
```
Comment (alan-agius4) [inline .ng-dev/format.mjs:12]: Use `.prettierignore` instead.

https://github.com/angular/angular-cli/blob/5226c535bd94bb6c57aa6305e1acef0a70095478/.prettierignore#L15

**Review by alan-agius4 — APPROVED** [review]
One small thing, otherwise looks good.

**Comment by crisbeto** [comment]
Feedback has been addressed.

**Pull request merged**